src/main/java/com/soboleoko/medicalclinic/service/VisitService.java

```diff
@@ -24,20 +23,18 @@ public class VisitService {
     private final DoctorRepository doctorRepository;
 
     @Transactional
-    public Visit createVisit(CreateVisitDTO createVisitDTO, Long doctorId) {
-        Visit visit = new Visit();
-        visit.setStartDate(createVisitDTO.getStartDate());
-        visit.setEndDate(createVisitDTO.getEndDate());
+    public Visit createVisit(Visit visit, Long doctorId) {
         Doctor doctor = doctorRepository.findById(doctorId)
                 .orElseThrow(() -> new DoctorNotFoundException("Doctor does not exist", HttpStatus.NOT_FOUND));
         visit.setDoctor(doctor);
-        if (createVisitDTO.getPatientId() != null) {
-            Patient patient = patientRepository.findById(createVisitDTO.getPatientId())
+        if (visit.getPatient() != null) {
+            Patient patient = patientRepository.findById(visit.getPatient().getId())
```

Comment:
mozesz wziac optionala i po prostu
visit.setPatient(optionalPatient.orElse(null)) to bedzie to samo a nie bedzie elsea